### ISSUE TITLE
FLAKY: bins/msan/h2_http_proxy_test stream_compression_compressed_payload GRPC_POLL_STRATEGY=epoll1 

### ISSUE THREAD

**Issue by muxi** [grpc/grpc#12307]
Happened in https://sponge.corp.google.com/target?id=b61b4754-bec5-41a2-9cc5-2d432eca8ce2&target=github/grpc/c_linux_msan_native&searchFor=&show=ALL&sortBy=STATUS.

**Comment by yashykt** [comment]
A similar bug with probably the same root cause http://sponge/aea7fb41-f89f-45c4-84af-50f9201e4115

**Comment by apolcyn** [comment]
tests failure (not a sanitizer failure) in the asan build of this too

```
E0915 02:20:14.609543658   13352 cq_verifier.c:280]          expected tags:
tag(101) GRPC_OP_COMPLETE success=1 test/core/end2end/tests/compressed_payload.c:194
```
https://grpc-testing.appspot.com/job/gRPC_pull_requests_asan_c/7116/testReport/junit/(root)/c_linux_asan_native/bins_asan_h2_http_proxy_test_compressed_payload_GRPC_POLL_STRATEGY_epoll1/

**Comment by muxi** [comment]
This should be resolved as part of #12399. Closing in a few days if it does not recur.
